Webui changes

viveksingh101994 opened the pull request:
Dashboard
<img width="1398" height="933" alt="image" src="https://github.com/user-attachments/assets/bae588e8-d40d-4f8d-a515-023a6f8e6df4" />

Connections
<img width="1386" height="886" alt="image" src="https://github.com/user-attachments/assets/57b3548c-1897-47ca-9bb1-0ea6bc912a2d" />

Graph
<img width="1388" height="912" alt="image" src="https://github.com/user-attachments/assets/646c48e1-01ec-46c5-a5a5-429e82f64448" />


Settings
<img width="1354" height="955" alt="image" src="https://github.com/user-attachments/assets/5ddd8e1c-c376-4a20-a577-7f056c6fd54d" />

viveksingh101994 committed:
chore: adds persona editor

viveksingh101994 committed:
remove: autosave

viveksingh101994 committed:
docs: adds docs file

viveksingh101994 committed:
chore: add file

viveksingh101994 committed:
chore: adds dashboard

viveksingh101994 committed:
chore: adds mcp card

viveksingh101994 committed:
chore: adds graph

viveksingh101994 committed:
chore: adds mock data

viveksingh101994 committed:
chore: adds ui change

viveksingh101994 committed:
chore: remove unused files

viveksingh101994 committed:
chore: updating the packages

viveksingh101994 committed:
updating to latest packages

viveksingh101994 committed:
Merge branch 'object-types' into webui-changes

TranHung92 commented on packages/client/package.json:
```diff
@@ -10,15 +10,28 @@
+    "react-hook-form": "^7.66.1",
+    "react-hot-toast": "^2.6.0",
+    "react-router-dom": "^7.9.6",
+    "reactflow": "^11.11.4",
```
latest version is actually [`@xyflow/react`](https://www.npmjs.com/package/@xyflow/react) package

viveksingh101994 replied:
resolves with https://github.com/reality-platforms/ebee-oss/pull/14/commits/546aa5d2da3375fdc1de7a22372440201dec3396

TranHung92 commented on packages/client/package.json:
these look unused
```
* @hookform/resolvers
* react-hook-form
* zod
```

viveksingh101994 replied:
resolves with https://github.com/reality-platforms/ebee-oss/pull/14/commits/546aa5d2da3375fdc1de7a22372440201dec3396

TranHung92 commented on packages/client/tailwind.config.js:
I think this file is no longer needed with the latest tailwind

viveksingh101994 replied:
resolves with https://github.com/reality-platforms/ebee-oss/pull/14/commits/546aa5d2da3375fdc1de7a22372440201dec3396

TranHung92 commented:
<img width="1262" height="756" alt="CleanShot 2025-11-29 at 02 47 55@2x" src="https://github.com/user-attachments/assets/2d598866-821f-4dba-94ec-5c7963e277df" />

<img width="1474" height="1332" alt="CleanShot 2025-11-29 at 02 48 22@2x" src="https://github.com/user-attachments/assets/1d611b6f-39a9-463e-97d1-b9a8d2fdaa36" />
Center labels look kinda odd. Left align would be better I think

![CleanShot 2025-11-29 at 02 50 36](https://github.com/user-attachments/assets/24168406-ffa5-456d-8d35-4b36bc2f1bad)
make top bar sticky. Remove unnecessary scrolling

viveksingh101994 committed:
chore: addressed the feedbacks

viveksingh101994 committed:
Merge branch 'object-types' of github.com:reality-platforms/ebee-oss into webui-changes

viveksingh101994 committed:
Merge branch 'object-types' of github.com:reality-platforms/ebee-oss into webui-changes

viveksingh101994 committed:
feat: add sync functionality with progress tracking to MCP servers

Added synchronization capability for MCP server configurations with visual progress feedback:

- Implemented useSyncMCPServer hook to handle sync mutations
- Added sync button to MCPServerCard that appears when server is connected
- Integrated progress bar showing sync status (0-100%)
- Added spinning RefreshCw icon during sync operations
- Included simulated progress updates for better user feedback
- Added success/error toast notifications for sync operations
- Automatically invalidates server queries after successful sync

The sync feature allows users to refresh MCP server configurations on-demand, improving data consistency and user control.

viveksingh101994 committed:
refactor: display server name instead of ID in sync toast messages

Update the sync mutation to accept both configId and server name,
allowing the success toast notification to display the user-friendly
server name instead of the technical configuration ID. This improves
the user experience by showing more meaningful feedback during the
sync operation.

Changes:
- Modified syncMutation call to pass server name alongside configId
- Updated useSyncMCPServer hook to accept name parameter
- Changed success toast message to display server name

TranHung92 commented on packages/client/src/lib/api.ts:
```diff
@@ -0,0 +1,284 @@
+      `/mcp-servers/${encodeURIComponent(name)}/status`
+    ),
+  sync: (configId: string) =>
+    api.post<ApiResponse<void>>(`/sync/${encodeURIComponent(configId)}`),
```
shall we change the endpoint to have config ids in the request body instead of params?

viveksingh101994 replied:
I would say request body would be better to make it use for multiple sync instead of just one. Lets say user wants to sync N number of MCP servers by selecting them on UI 🤔

viveksingh101994 replied:
https://github.com/reality-platforms/ebee-oss/pull/14/commits/d73abae0f3313b1d4100625d0e6174b3341d456a

viveksingh101994 committed:
chore: changes the signature of api to body

viveksingh101994 committed:
chore: adds graph view from graphstore

viveksingh101994 committed:
add graph indexing

viveksingh101994 committed:
chore(logging): remove SSE debug console logs

Remove console.log statements from sync progress SSE implementation in both client and server packages. These debug logs were used during development and are no longer needed in production code.

Changes:
- Remove SSE progress update logging from useSyncProgress hook (client)
- Remove job state and progress logging from server SSE endpoint

buildgreatthings commented on docker-compose.yml:
```diff
@@ -28,6 +28,20 @@ services:
     networks:
       - ebee-network
 
+  memgraph_ui:
```
remove this

buildgreatthings commented on docker-compose.yml:
```diff
@@ -54,6 +68,50 @@ services:
+  #     - ebee-network
+  #   command: pnpm --filter server run dev
+
+  # client:
```
enable this

viveksingh101994 committed:
chore: remove memgraph-ui

viveksingh101994 committed:
feat(graph): add multi-directional handles to graph nodes

Add Handle components to DataNode for better edge connections:
- Add target handles on top and left positions
- Add source handles on right and bottom positions
- Remove hardcoded sourcePosition/targetPosition from node config
- Set data view as default mode and hide view toggle buttons
- Improve UI spacing and heading size

This allows nodes to connect from multiple directions instead of being restricted to right/left connections, providing more flexible graph layouts.

viveksingh101994 committed:
Merge branch 'object-types' of github.com:reality-platforms/ebee-oss into webui-changes

viveksingh101994 committed:
feat(indexing): skip re-indexing vectors with unchanged checksums

Add checksum-based deduplication to avoid re-indexing records whose
vectors are already up-to-date in the vector store. This improves
indexing performance by preventing redundant embedding operations.

Changes:
- Add hasPointsWithChecksum() method to VectorStore to check if vectors
  exist for a given mongoId and checksum
- Check for existing vectors before processing in insertRecordToVectorDB()
- Track skipped records in indexing statistics
- Update progress logging to include skipped record count

This optimization prevents unnecessary re-indexing when content hasn't
changed, reducing API calls and processing time during bulk operations.

buildgreatthings commented on packages/server/src/mock/README.md:
```diff
@@ -0,0 +1,180 @@
+# Mock Data Feature
```
rip this out

viveksingh101994 replied:
removed it

viveksingh101994 committed:
chore: removing the mock data

viveksingh101994 committed:
chore: remove server side code

viveksingh101994 committed:
chore: fix the stats

viveksingh101994 committed:
chore: remove refresh button

viveksingh101994 committed:
chore: reverting the other file changes for store

viveksingh101994 committed:
chore: revert function"

viveksingh101994 committed:
chore: adding stats increment code

viveksingh101994 committed:
chore: remove SSE

viveksingh101994 committed:
chore: removing env
